analysis/analysis.js

```diff
@@ -0,0 +1,163 @@
+
+var http = require('http');
+var mongoose = require('mongoose');
+var request = require('request');
+var Qs = require('qs');
+var fs = require('fs');
+
+// var rstats  = require('rstats');
+// var R  = new rstats.session();
+// var schemas = {
+//   analysisData: require('./model/analysisData').schema,
+// };
+var analysisInprogress = true;
+
+function analyzeFrame(rideId, frame){
+  console.log("IN ANALYSIS FUNCTION");
+  var timeAtEvent = frame[50].timestamp;
+  var requestData = {
+    'violationType': 'Speedbreaker',
+    'geo': {
+      'latitude': frame[50].geo.latitude,
+      'longitude': frame[50].geo.longitude,
+      'accuracy': frame[50].geo.accuracy,
+      'altitude': frame[50].geo.altitude,
+      'altitudeAccuracy': frame[50].geo.altitudeAccuracy,
+      'heading': frame[50].geo.heading,
+      'speed': frame[50].geo.speed*3.6,
+    },
+    'severity': 1,
+    'description': "This is a speedbreaker"
+  };
+    return requestData;
+};
+
+function analyzeRide(){
+  if(!analysisInprogress){
```

Comment:
you can instead do `if(analysisInProgress) return;`. This can save the indentation depth.